6-Typescript/server/controllers/todoController.ts

```diff
@@ -1,78 +1,66 @@
-import { ITodo } from './../interfaces/ITodo';
-import {Request, Response} from 'express'
-import { ITodoService } from '../interfaces/ITodoService';
+import { ITodo } from "../interfaces/interfaces";
+import { Request, Response } from "express";
+import { ITodoService } from "../interfaces/interfaces";
 export class TodoController {
-    db: ITodoService
+    private db: ITodoService;
     constructor(db: ITodoService) {
-        this.db = db
-
+        this.db = db;
     }
     getTodoById = async (req: Request, res: Response): Promise<void> => {
         const userId: string = req.body.userId;
-        const todoId: string= req.params.id
-        
-        const todo: ITodo | null = await this.db.modelGetTodoById(userId, todoId);
-        if(todo) {
-            res.status(200).json(todo)
-        } else {                                                
-            res.sendStatus(404)
+        const todoId: string = req.params.id;
+
+        const todo: ITodo | null = await this.db.modelGetTodoById(
+            userId,
+            todoId
+        );
+        if (todo == null) {
+            res.sendStatus(404);
+            return;
         }
-    }
-    postTodo = async (req: Request, res: Response): Promise<void> => {   
+        res.status(200).json(todo);
+    };
+    postTodo = async (req: Request, res: Response): Promise<void> => {
+        const newTodo = req.body;
         try {
-            const {userId, todoText, todoId, checked}:
-            { userId:string, todoText:string, todoId:string, checked:boolean }
-             = req.body
-            const newTodo: ITodo = {
-                todoText: todoText,
-                todoId: todoId,
-                checked: checked,
-                userId: userId,
-            }
             const todo: ITodo | null = await this.db.modelPostTodo(newTodo);
-            if(!todo) {
-                res.status(400).send('Failed to post todo');
-            } else {
-                res.sendStatus(201);
+            res.sendStatus(201);
+        } catch (err: any) {
```

Comment:
why `any`?

Reply:
when i tried to give it a type, i got the following error:
"Catch clause variable type annotation must be 'any' or 'unknown' if specified"

I can the unknown as well.
What do you think?

Reply:
yes, you're right.
but then you try to access `err.name` which is maybe not exist...
you can use `instanceOf` before accessing your error
something like 
```ts
catch(err){
if (err isntanceof YuvalErrorType){
err.name = {........}
}
}
```

references:
https://fettblog.eu/typescript-typing-catch-clauses/#2.-there-is-only-one-catch-clause-in-javascript 
https://stackoverflow.com/questions/45379959/narrowing-down-error-type-in-catch